File: sortowanie.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
#include <cstdlib>
#include <ctime>
#include <chrono>
#include <thread>
#include <string>

#ifdef _WIN32
#include <Windows.h>
#else
#include <sys/sysinfo.h>
#endif

// Funkcja do obliczania sumy
long long some_function() {
    long long total = 0;
    for (int i = 1; i < 1000000; ++i) {
        total += i;
    }
    return total;
}

// Funkcja do uzyskania informacji o CPU w systemie Linux
void get_cpu_info(std::ofstream &file) {
#ifdef _WIN32
    SYSTEM_INFO sysInfo;
    GetSystemInfo(&sysInfo);
    file << "Liczba rdzeni CPU (fizycznych): " << sysInfo.dwNumberOfProcessors << std::endl;
#else
    file << "Liczba rdzeni CPU (fizycznych): " << std::thread::hardware_concurrency() << std::endl;
#endif
}

// Funkcja do uzyskania informacji o pamięci w systemie Linux
void get_memory_info(std::ofstream &file) {
#ifdef _WIN32
    MEMORYSTATUSEX statex;
    statex.dwLength = sizeof(statex);
    GlobalMemoryStatusEx(&statex);
    file << "Całkowita pamięć RAM: " << statex.ullTotalPhys << " bytes\n";
    file << "Wykorzystana pamięć RAM: " << statex.ullTotalPhys - statex.ullAvailPhys << " bytes\n";
    file << "Wolna pamięć RAM: " << statex.ullAvailPhys << " bytes\n";
#else
    struct sysinfo memInfo;
    sysinfo(&memInfo);
    file << "Całkowita pamięć RAM: " << memInfo.totalram << " bytes\n";
    file << "Wykorzystana pamięć RAM: " << (memInfo.totalram - memInfo.freeram) << " bytes\n";
    file << "Wolna pamięć RAM: " << memInfo.freeram << " bytes\n";
#endif
}

int main() {
    std::string filepath = "dane.txt";
    std::ifstream file(filepath);
    std::vector<double> lines;
    double x;
    
    while (file >> x) {
        lines.push_back(x);
    }

    auto start_time = std::chrono::high_resolution_clock::now();

    std::sort(lines.begin(), lines.end());

    auto end_time = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> execution_time = end_time - start_time;

    std::string sorted_filepath = "posortowane_dane.txt";
    std::ofstream sorted_file(sorted_filepath);
    for (auto &value : lines) {
        sorted_file << value << "\n";
    }

    std::string wynik_filepath = "wynik.txt";
    std::ofstream wynik_file(wynik_filepath);
    wynik_file << "Czas sortowania: " << execution_time.count() << " sekundy\n";
    wynik_file << "Program wykonujący sortowanie: C++\n";

    get_cpu_info(wynik_file);
    get_memory_info(wynik_file);

    // Dodatkowe informacje o systemie
    wynik_file << "System operacyjny: ";
#ifdef _WIN32
    wynik_file << "Windows\n";
#else
    wynik_file << "Linux\n";
#endif

    std::cout << "Informacje zostały zapisane do pliku " << wynik_filepath << "." << std::endl;

    return 0;
}

File: sortowanie.py
import datetime
import sys
import psutil
import platform

def some_function():
    total = 0
    for i in range(1, 1000000):
        total += i
    return total

# Zakładając, że filepath jest zdefiniowane w module 'generowanie'
from generowanie import filepath

# Wczytanie danych z pliku
with open('dane.txt', 'r') as file:
    lines = file.readlines()

# Konwersja danych do float
lines = [float(line.strip()) for line in lines]

# Mierzenie czasu sortowania
start_time = datetime.datetime.now()
posortowane = sorted(lines)
end_time = datetime.datetime.now()

# Czas wykonania sortowania
execution_time = end_time - start_time

# Zapisanie posortowanych danych do pliku
sorted_filepath = "posortowane_dane.txt"
with open(sorted_filepath, 'w') as file:
    for x in posortowane:
        file.write(f"{x}\n")

# Zapisanie informacji o czasie i systemie do oddzielnego pliku
wynik_filepath = "wynik.txt"  # Plik, do którego zapiszemy dodatkowe informacje
with open(wynik_filepath, 'w') as file:
    file.write(f"Czas sortowania: {execution_time}\n")
    file.write(f"Program wykonujący sortowanie: Python {sys.version}\n")

    # Informacje o CPU
    file.write(f"Liczba rdzeni CPU (fizycznych): {psutil.cpu_count(logical=False)}\n")  # Liczba rdzeni fizycznych
    file.write(f"Liczba logicznych rdzeni CPU: {psutil.cpu_count(logical=True)}\n")  # Liczba rdzeni logicznych
    file.write(f"Obciążenie CPU: {psutil.cpu_percent(interval=1)}%\n")  # Obciążenie CPU w procentach

    # Informacje o pamięci RAM
    mem = psutil.virtual_memory()
    file.write(f"Całkowita pamięć RAM: {mem.total} bytes\n")
    file.write(f"Wykorzystana pamięć RAM: {mem.used} bytes\n")
    file.write(f"Wolna pamięć RAM: {mem.free} bytes\n")

    # Informacje o dysku
    disk = psutil.disk_usage('/')
    file.write(f"Całkowita pamięć dysku: {disk.total} bytes\n")
    file.write(f"Wykorzystana pamięć dysku: {disk.used} bytes\n")
    file.write(f"Wolna pamięć dysku: {disk.free} bytes\n")

    # Informacje o systemie
    file.write(f"Wersja systemu: {platform.version()}\n")
    file.write(f"Nazwa systemu operacyjnego: {platform.system()}\n")

print(f"Informacje zostały zapisane do pliku {wynik_filepath}.")
